redhat-appstudio/service-provider-integration-operator
# Always update the upload URL in the SPIAccessToken status

### What does this PR do?
Make sure to regenerate the upload URL just in case of BASE_URL changes.

### What issues does this PR fix or reference?
none

### How to test this PR?
1. `make install deploy_minikube SPIO_IMG=quay.io/lkrejci/service-provider-integration-operator:upload-url-fix`
2. `kubectl apply -f samples/binding-read-repo.yaml`
3. Check the upload URL: `kubectl get spiaccesstokenbinding binding-read-repository -ojsonpath='{.status.uploadUrl}`
5. `kubectl -n spi-system patch configmap/spi-shared-environment-config --type=json -p='[{"op": "replace", "path": "/data/BASEURL", "value": "https://now.its.there"}]'`
6. `kubectl -n spi-system rollout restart deployment/spi-controller-manager`
3. Check the upload URL: `kubectl get spiaccesstokenbinding binding-read-repository -ojsonpath='{.status.uploadUrl}`
